JASP-Engine/JASP/R/regressionloglinearbayesian.R

```diff
@@ -16,355 +16,521 @@
 #
 
 RegressionLogLinearBayesian <- function(dataset, options, perform="run", callback, ...) {
-    counts.var <- options$counts
-	if (counts.var == "") {
-	    counts.var <- NULL
-	}
-		
-	if (is.null(dataset)) {
+    # Init end result
+    results <- list()
+    meta <- list()
+    .meta <-  list(list(name = "title", type = "title"),
+                   list(name = "table", type = "table"),
+                   list(name = "posteriorTable", type = "table"),
+                   list(name = "Bayesianlogregression", type = "table"),
+                   list(name = "BayesianSublogregression", type = "table")
+    )
+    
+    
+    results[[".meta"]] <- .meta
+    results[["title"]] <- "Bayesian Log-Linear Regression"
+    
+    logLinearBayesianCitations <- 	list(
+        "Overstall, A., & King, R. (2014). conting: an R package for Bayesian analysis of complete and incomplete contingency tables. Journal of Statistical Software, 58(7), 1-27."
+    )
+    
+    # Init table
+    posteriorTable <- list()
+    
+    posteriorTable[["title"]] <- "Model Comparison"
+    posteriorTable[["citation"]] <- logLinearBayesianCitations
+    
+    if (options$bayesFactorType == "BF10") {
+        bfTitle <- "BF<sub>10</sub>"
+    } else if (options$bayesFactorType == "BF01") {
+        bfTitle <- "BF<sub>01</sub>"
+    } else {
+        bfTitle <- "Log(BF<sub>10</sub>)"
+    }
+    
+    # Declare table elements
+    fields <- list(
```

Comment:
Seems odd to specify this here. I guess it's because this entire analysis is one big function. Would be very nice to split it into smaller dedication functions (calculate, build table, etc) at some point.